netbeans/CheckerFrameworkPlugin/src/org/checkerframework/netbeans/CheckerFrameworkPanel.java

```diff
@@ -0,0 +1,159 @@
+package org.checkerframework.netbeans;
+
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map.Entry;
+import java.util.Set;
+import javax.swing.BoxLayout;
+import javax.swing.JCheckBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import org.netbeans.api.project.ProjectManager;
+import org.netbeans.spi.project.support.ant.EditableProperties;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.modules.InstalledFileLocator;
+import org.openide.util.MutexException;
+
+public class CheckerFrameworkPanel extends JPanel {
+
+    private EditableProperties editableProperty;
+    private static Set<Entry<String, String>> checkerStrings;
+    private FileObject projectProperties;
+
+    private JLabel title;
+    private JCheckBox[] checkerList;
+    private StringBuilder selection;
+    private final String checkerPath;
+    private final String checkerQualPath;
+
+    /**
+     * Constructor for the Checker Framework panel in the Project Properties window.
+     *
+     * @param inProjectProperties A file object containing the Project Properties
+     */
+    public CheckerFrameworkPanel(FileObject inProjectProperties) {
+        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+
+        title = new JLabel("Run Built-In Checker");
+        this.add(title);
+
+        projectProperties = inProjectProperties;
+        checkerPath =
+                InstalledFileLocator.getDefault()
+                        .locate("checker.jar", "org.checkerframework.netbeans", false)
+                        .getAbsolutePath();
+        checkerQualPath =
+                InstalledFileLocator.getDefault()
+                        .locate("checker-qual.jar", "org.checkerframework.netbeans", false)
+                        .getAbsolutePath();
+
+        try {
+            editableProperty = loadProperties(projectProperties);
+        } catch (IOException e) {
+            System.out.println("Failed to load netbeans project.properties file.");
+        }
+
+        try {
+            checkerStrings =
+                    loadProperties(
+                                    FileUtil.toFileObject(
+                                            InstalledFileLocator.getDefault()
+                                                    .locate(
+                                                            "checkerstrings.properties",
+                                                            "org.checkerframework.netbeans",
+                                                            false)))
+                            .entrySet();
+        } catch (IOException e) {
+            System.out.println("Failed to load checker strings properties file.");
+        }
+
+        checkerList = new JCheckBox[checkerStrings.size()];
```

Comment:
If the strings property file failed to load, you'll get a NullPointerException here, because the catch block simply outputs an error and then continues.